Composer/packages/client/src/pages/language-generation/table-view.js

```diff
@@ -31,13 +31,18 @@ export default function TableView(props) {
 
   useEffect(() => {
     if (isEmpty(lgFile)) return;
-
-    const allTemplates = lgUtil.parse(lgFile.content).map((template, templateIndex) => {
-      return {
-        ...template,
-        index: templateIndex,
-      };
-    });
+    let allTemplates = [];
+    try {
+      allTemplates = lgUtil.parse(lgFile.content).map((template, templateIndex) => {
+        return {
+          ...template,
+          index: templateIndex,
+        };
+      });
+      // mute lg file invalid cause page crash, setState is async, this component may render at first
+    } catch (error) {
+      console.error(error);
```

Comment:
When LG parser failed to parse a file, the error message should contains the diagnostics information to tell us what's wrong, we should direct user to edit mode (in which Manaco will also give out the sample message) to fix that, right?

Reply:
if parsing failed, error is handled by upon component, which will navigate to edit mode, and show error directly in monaco editor.
Here try parse is a trick for `setEditMode`, react state set not synchronize change, this `table-view` component may mounted at first time.